src/transformers/generation_tf_utils.py

```diff
@@ -756,6 +759,8 @@ def generate(
             )
             # expand encoder_outputs
             encoder_outputs = (tf.gather(encoder_outputs[0], expanded_batch_idxs, axis=0),)
+            if "attention_mask" in locals():  # vision models don't have this
```

Comment:
Excellent point, for some reason I didn't notice that it was in the keyword arguments (and thus the current version would fail) 👍  Checking against `None` now